src/zero_copy/combinator.rs

```diff
@@ -81,51 +102,56 @@ where
         })
     }
 
-    go_extra!();
+    go_extra!(O);
 }
 
 #[derive(Copy, Clone)]
-pub struct Map<A, F> {
+pub struct Map<A, OA, F> {
     pub(crate) parser: A,
     pub(crate) mapper: F,
+    // FIXME: try remove 'OA' type parameter?
+    // This phantom seems necessary to have OA as part of impl<.., OA>.
+    // Otherwise it looks like: impl<.., OA> Parser<..> for Map<A, F>
+    // and I get the error [E0207]:
+    // `the type parameter `OA` is not constrained by the impl trait, self type, or predicates`
+    // For some reason it doesn't see that OA is used for 'A', which is mentioned for Map<..>.
```

Comment:
This change is correct, and unfortunately required. You can see it in `master`'s definition of [`Map`](https://docs.rs/chumsky/latest/chumsky/combinator/struct.Map.html). No need for the comment.